docs/howto/garbage-collection/standalone-gc.md

```diff
@@ -0,0 +1,189 @@
+   ```
+
+Example (minimalistic) config file:
+```yaml
+logging:
+  level: debug
+lakefs:
+  endpoint_url: https://your.url/api/v1
+  access_key_id: <your lakeFS access key>
+  secret_access_key: <your lakeFS secret key>
+```
+
+### Command line reference
+
+#### Flags:
+- `-c, --config`: config file to use (default is $HOME/.lakefs-sgc.yaml)
+
+#### Commands:
+**run**
+
+Usage: \
+`lakefs-sgc run <repository>`
+
+Flags:
+- `--cache-dir`: directory to cache read files and metadataDir (default is $HOME/.lakefs-sgc/data/)
+- `--parallelism`: number of parallel downloads for metadataDir (default 10)
+- `--presign`: use pre-signed URLs when downloading/uploading data (recommended) (default true)
+
+### Example - docker run command
+Here's an example for running the `treeverse/lakefs-sgc` docker image, with AWS credentials parsed from the `~/.aws/credentials` file:
+
+```bash
+docker run \
+-e AWS_REGION=<region> \
+-e AWS_SESSION_TOKEN="$(grep 'aws_session_token' ~/.aws/credentials | awk -F' = ' '{print $2}')" \
+-e AWS_ACCESS_KEY_ID="$(grep 'aws_access_key_id' ~/.aws/credentials | awk -F' = ' '{print $2}')" \
+-e AWS_SECRET_ACCESS_KEY="$(grep 'aws_secret_access_key' ~/.aws/credentials | awk -F' = ' '{print $2}')" \
+-e LAKEFS_SGC_LAKEFS_ENDPOINT_URL=<your lakefs URL> \
+-e LAKEFS_SGC_LAKEFS_ACCESS_KEY_ID=<your accesss key> \
+-e LAKEFS_SGC_LAKEFS_SECRET_ACCESS_KEY=<your secret key> \
+-e LAKEFS_SGC_LOGGING_LEVEL=debug \
+treeverse/lakefs-sgc:<version> run <repository>
+```
+
+### Output
+`lakefs-sgc` will write its reports to `<REPOSITORY_STORAGE_NAMESPACE>/_lakefs/retention/gc/reports/<RUN_ID>/`. \
+_RUN_ID_ is generated during runtime by the Standalone GC. You can find it in the logs:
+```
+"Marking objects for deletion" ... run_id=gcoca17haabs73f2gtq0
+```
+
+In this prefix, you'll find 3 objects:
+- `deleted.json` - Containing all marked objects in a json format. 
+- `deleted.parquet` - Containing all marked objects in a parquet format, with the following schema:
+   ```
+  ┌─────────────┬─────────────┬─────────┬─────────┬─────────┬─────────┐
+  │ column_name │ column_type │  null   │   key   │ default │  extra  │
+  │   varchar   │   varchar   │ varchar │ varchar │ varchar │ varchar │
+  ├─────────────┼─────────────┼─────────┼─────────┼─────────┼─────────┤
+  │ address     │ VARCHAR     │ YES     │         │         │         │
+  └─────────────┴─────────────┴─────────┴─────────┴─────────┴─────────┘
+  ```
+- `summary.json` - A small json summarizing the GC run. Example:
+```json
+{
+    "run_id": "gcoca17haabs73f2gtq0",
+    "success": true,
+    "first_slice": "gcss5tpsrurs73cqi6e0",
+    "start_time": "2024-10-27T13:19:26.890099059Z",
+    "cutoff_time": "2024-10-27T07:19:26.890099059Z",
+    "num_deleted_objects": 33000
+}
+```
+
+### Deleting marked objects
+To delete the objects marked by the GC, you'll need to read the `deleted.parquet` or `deleted.json` files, and manually delete each address from AWS.
```

Comment:
Please provide a bash command/script to:
1. Parse the deleted objects and delete them
2. Move them to somewhere else in the storage namespace (i.e. safe delete) and a way to restore them if needed.

Reply:
Done - added an example for parsing the addresses and moving them to another bucket, the leap from here to `aws s3 rm` is trivial.